app/controllers/CyaMaintainProfileController.scala

```diff
@@ -138,21 +138,28 @@ class CyaMaintainProfileController @Inject() (
       }
     }
 
-  def onPageLoadNirmsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request =>
-      TraderProfile.getOptionallyRemovedPage(request.userAnswers) match {
-        case Right(_)     =>
-          val list = SummaryListViewModel(
-            rows = Seq(
-              HasNirmsSummary.rowUpdate(request.userAnswers),
-              NirmsNumberSummary.rowUpdate(request.userAnswers)
-            ).flatten
-          )
-          Future.successful(Ok(view(list, routes.CyaMaintainProfileController.onSubmitNirmsNumber)))
-        case Left(errors) =>
-          Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
+  def onPageLoadNirmsNumber(): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
+      (for {
+        nirmsNumberOpt <- traderProfileConnector.getTraderProfile(request.eori).map(_.nirmsNumber)
```

Comment:
pedantic, might be slightly more readable/debuggable to split this out to like 

        traderProfile <- traderProfileConnector.getTraderProfile(request.eori)
        nirmsNumberOpt = traderProfile.nirmsNumber